core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala

```diff
@@ -105,7 +105,8 @@ private[spark] class FileShuffleBlockResolver(conf: SparkConf)
    * when the writers are closed successfully
    */
   def forMapTask(shuffleId: Int, mapId: Int, numBuckets: Int, serializer: Serializer,
-      writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
+      writeMetrics: ShuffleWriteMetrics,
+      getDiskWriter: (BlockId, File, SerializerInstance, Int, ShuffleWriteMetrics) => BlockObjectWriter = blockManager.getDiskWriter): ShuffleWriterGroup = {
```

Comment:
This isn't a Java-friendly interface, which is going to be a problem for new shuffle code that I'm working on.